### PR TITLE
mesa wayland: Fix/simplify cross

### LINKED ISSUE #88961: mesa fails to (cross-)compile
**Describe the bug**

Cross-compilation of mesa fails with the following error at configure stage

```
$ nix build -L -f '<nixpkgs>' pkgsCross.aarch64-multiplatform.mesa
...
mesa> Run-time dependency valgrind found: YES 3.15.0
mesa> Program bison found: YES (/nix/store/ra42sij0cb917p3m90qdnv0vc3kxv1hm-bison-3.5.4/bin/bison)
mesa> Program flex found: YES (/nix/store/q99r9k0py7m2mrdfsif9wfvllvvxfiqi-flex-2.6.4/bin/flex)
mesa> Run-time dependency libunwind found: NO (tried pkgconfig and cmake)
mesa> Found pkg-config: /nix/store/nc3ivxhlqyj2alck8p47srvwzzgkbh2i-pkg-config-0.29.2/bin/pkg-config (0.29.2)
mesa> Did not find CMake 'cmake'
mesa> Found CMake: NO
mesa> Build-time dependency wayland-scanner found: NO (tried pkgconfig)
mesa> meson.build:1553:2: ERROR: Dependency "wayland-scanner" not found, tried pkgconfig
mesa> A full log can be found at /build/mesa-20.0.2/build/meson-logs/meson-log.txt
```
Though there appear to be other non-fatal, but otherwise concerning issues (especially surrounding LLVM) in the configuration process.

**To Reproduce**
1. `nix build -L -f '<nixpkgs>' pkgsCross.aarch64-multiplatform.mesa`

**Expected behavior**
Build should succeed

**Notify maintainers**

@vcunat

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.13, NixOS, 20.09pre225898.7319061eef4 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.09pre225898.7319061eef4"`
 - channels(nagisa): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Fix #88961

CC @nagisa

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nagisa** [comment]
LGTM, will try it out in a sec.

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -73,6 +73,12 @@ stdenv.mkDerivation {
       })
     ];
 
+  postPatch = ''
+    substituteInPlace meson.build --replace \
```
Comment (matthewbauer) [inline pkgs/development/libraries/mesa/default.nix:77]: doesn't meson know of two pkg-config versions?

Comment (Ericson2314) [reply pkgs/development/libraries/mesa/default.nix:77]: Meson itself does, but for some reason this is using a generic find-program that doesn't know about the program it's looking for.

**Comment by nagisa** [comment]
Can confirm that this has the intended effect and mesa now does in fact properly cross-compile.

**Pull request merged**